pip_audit/_dependency_source/pip.py

```diff
@@ -83,13 +83,17 @@ def __init__(
         effective_python = os.environ.get("PIPAPI_PYTHON_LOCATION", sys.executable)
         venv_prefix = os.getenv("VIRTUAL_ENV")
         if venv_prefix is not None and not effective_python.startswith(venv_prefix):
-            logger.warning(
-                f"pip-audit will run pip against {effective_python}, but you have "
-                f"a virtual environment loaded at {venv_prefix}. This may result in "
-                "unintuitive audits, since your local environment will not be audited. "
-                "You can forcefully override this behavior by setting PIPAPI_PYTHON_LOCATION "
-                "to the location of your virtual environment's Python interpreter."
-            )
+            venv_python = Path(venv_prefix + "/bin/python3")
+            if venv_python.exists():
+                os.environ["PIPAPI_PYTHON_LOCATION"] = str(venv_python)
+            else:
+                logger.warning(
+                    f"pip-audit will run pip against {effective_python}, but you have "
```

Comment:
I thought about editing this to include the fact that we tried to look for a symlink in `bin/` but wasn't sure how to phrase it as this is already getting complicated. I don't think it's all that important to communicate to the user since the corrective action doesn't change.

Reply:
We could also maybe just fail here?

Reply:
Yeah, I think just failing is probably appropriate -- the lack of a `bin` here suggests that the user intended the load a virtual environment (based on the env setting) but somehow ended up with a broken on, so we shouldn't try to guess at the correct behavior.